### PR TITLE
ocamlPackages.capnp: init at 3.6.0 

### PR DESCRIPTION
**Pull request by Sixstring982**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/ocaml-modules/res/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib , fetchFromGitHub , buildDunePackage }:
+
+buildDunePackage rec {
+  pname = "res";
+  version = "5.0.1";
+
+  minimalOCamlVersion = "4.08";
+
+  src = fetchFromGitHub {
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/res/default.nix:9]: You may want to get the source from there instead: https://github.com/mmottl/res/releases/download/5.0.1/res-5.0.1.tbz

File: pkgs/top-level/ocaml-packages.nix [hunk 2]
```diff
@@ -1608,6 +1610,8 @@ let
 
     resource-pooling = callPackage ../development/ocaml-modules/resource-pooling { };
 
+    res = callPackage ../development/ocaml-modules/res { };
+
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix:1614]: Please mind the alphabetical order.

Comment (Sixstring982) [reply pkgs/top-level/ocaml-packages.nix:1612]: Whoops -- fixed!

File: pkgs/development/ocaml-modules/res/default.nix (hunk 1 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/res/default.nix]: Sure, sounds good -- I'll use `fetchurl` instead here.

File: pkgs/development/ocaml-modules/capnp/default.nix [hunk 3]
```diff
@@ -0,0 +1,50 @@
+{ pkgs
+, lib
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/capnp/default.nix:2]:
```suggestion
{ lib
```

This parameter is not used.

File: pkgs/development/ocaml-modules/capnp/default.nix [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    rev = "v${version}";
+    hash = "sha256-G4B1llsHnGcuGIarDB248QMaRBvS47IEQB5B93wY7nA=";
+  };
+
+  nativeBuildInputs = [
+    capnproto
+  ];
+
+  buildInputs = [
+    result
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/capnp/default.nix:31]: The `result` library belongs to `propagatedBuildInputs`.

File: pkgs/development/ocaml-modules/capnp/default.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    stdio
+  ];
+
+  propagatedBuildInputs = [
+    ocplib-endian
+    res
+    stdint
+  ];
+
+  doCheck = false;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/capnp/default.nix:41]: This line without a comment is surprising. Are tests costly, broken, irrelevant?

File: pkgs/development/ocaml-modules/capnp/default.nix [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+  meta = {
+    description = "OCaml code generation plugin for the Cap'n Proto serialization framework";
+    homepage = "https://github.com/capnproto/capnp-ocaml";
+    changelog = "https://github.com/capnproto/capnp-ocaml/blob/${version}/CHANGES.md";
+    license = lib.licenses.bsd3;
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/capnp/default.nix:47]:
```suggestion
    license = lib.licenses.bsd2;
```

**Review by vbgl — COMMENTED** [review]
`res` LGTM, but `capnp` needs a bit of polishing.

File: pkgs/development/ocaml-modules/capnp/default.nix (hunk 3 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix:2]: Sounds good! Removed.

File: pkgs/development/ocaml-modules/capnp/default.nix (hunk 4 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix:31]: Sounds good -- fixed!

I have a hard time figuring out which build inputs need to be propagated.

How did you figure this out?

File: pkgs/development/ocaml-modules/capnp/default.nix (hunk 5 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix:41]: Hmm I'm not sure why. I think these tests seemed a little difficult to get working.

But they weren't -- looks like enabling them wasn't so bad!
They seem like they're pretty fast, too.

File: pkgs/development/ocaml-modules/capnp/default.nix (hunk 6 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix:47]: Oops, you're right! Fixed.

File: pkgs/development/ocaml-modules/capnp/default.nix (hunk 4 above)
Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix]: Excellent! Thanks for the tip!

File: pkgs/development/ocaml-modules/capnp/default.nix [hunk 7]
```diff
@@ -0,0 +1,56 @@
+{ lib
+, buildDunePackage
+, fetchFromGitHub
+, base_quickcheck
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/capnp/default.nix:4]:
```suggestion
, base_quickcheck ? null
```

Technical detail to fix evaluation.

Comment (Sixstring982) [reply pkgs/development/ocaml-modules/capnp/default.nix]: Sounds good -- done!

**Comment by sktrent** [comment]
Thanks for the thorough review, @vbgl!

**Pull request merged**